pal/screen-adapter/web/screen-adapter.ts

```diff
@@ -392,21 +402,53 @@ class ScreenAdapter extends EventTarget {
                 }, { once: true });
             };
             updateDPRChangeListener();
-        }
-        window.addEventListener('orientationchange', (): void => {
-            if (this._orientationChangeTimeoutId !== -1) {
-                clearTimeout(this._orientationChangeTimeoutId);
-            }
-            this._orientationChangeTimeoutId = setTimeout((): void => {
-                if (!this.handleResizeEvent) {
-                    return;
+            
+            const mediaQueryPortrait = window.matchMedia('(orientation: portrait)');
+            const mediaQueryLandscape = window.matchMedia('(orientation: landscape)');
+            const handleOrientationChange = (): void => {
+                let tmpOrientation = adapter._orientation;
+                const orientationType = screen.orientation.type;
+                if (mediaQueryPortrait.matches) {
+                    if (orientationType === 'portrait-primary') {
+                        tmpOrientation = Orientation.PORTRAIT;
+                    } else {
+                        tmpOrientation = Orientation.PORTRAIT_UPSIDE_DOWN;
+                    }
+                } else {
+                    if (orientationType === 'landscape-primary') {
+                        tmpOrientation = Orientation.LANDSCAPE_LEFT;
+                    } else if (orientationType === 'landscape-secondary') {
+                        tmpOrientation = Orientation.LANDSCAPE_RIGHT;
+                    }
                 }
-                this._updateFrameState();
-                this._resizeFrame();
-                this.emit('orientation-change', this.windowSize.width, this.windowSize.height);
-                this._orientationChangeTimeoutId = -1;
-            }, EVENT_TIMEOUT);
-        });
+                notifyOrientationChange(tmpOrientation);
+            };
+            mediaQueryPortrait.addEventListener('change', handleOrientationChange);
+            mediaQueryLandscape.addEventListener('change', handleOrientationChange);
+        } else {
+            const adapter = this;
+            const handleOrientationChange = (): void => {
+                let tmpOrientation = adapter._orientation;
+                switch (window.orientation) {
+                    case 0:
+                        tmpOrientation = Orientation.PORTRAIT;
+                        break;
+                    case 90:
+                        // Handle landscape orientation, top side facing to the right
+                        tmpOrientation = Orientation.LANDSCAPE_RIGHT;
+                        break;
+                    case -90:
+                        // Handle landscape orientation, top side facing to the left
+                        tmpOrientation = Orientation.LANDSCAPE_LEFT;
+                        break;
+                    default:
+                        tmpOrientation = Orientation.PORTRAIT_UPSIDE_DOWN;
```

Comment:
Why is the default value assigned with `PORTRAIT_UPSIDE_DOWN`?

Reply:
Only four values, so set the forth value to PORTRAIT_UPSIDE_DOWN

Reply:
It's better to check the fourth value  explicitly.